src/Extensions/Diagnostics.HealthChecks/Internal/OmexHealthCheckPublisher.cs

```diff
@@ -11,32 +11,36 @@
 
 namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks
 {
-	internal abstract class HealthCheckPublisher : IHealthCheckPublisher
+	internal class OmexHealthCheckPublisher : IHealthCheckPublisher
 	{
-		protected readonly ObjectPool<StringBuilder> StringBuilderPool;
+		protected readonly ObjectPool<StringBuilder> StringBuilderPool;
 
-		internal abstract string HealthReportSourceId { get; }
+		private IHealthStatusSender StatusSender { get; }
 
-		internal const string HealthReportSummaryProperty = "HealthReportSummary";
+		internal const string HealthReportSummaryProperty = "HealthReportSummary";
 
-		public HealthCheckPublisher(ObjectPoolProvider objectPoolProvider) => StringBuilderPool = objectPoolProvider.CreateStringBuilderPool();
+		public OmexHealthCheckPublisher(IHealthStatusSender statusSender, ObjectPoolProvider objectPoolProvider)
+		{
+			StringBuilderPool = objectPoolProvider.CreateStringBuilderPool();
+			StatusSender = statusSender;
+		}
 
-		public abstract Task PublishAsync(HealthReport report, CancellationToken cancellationToken);
+		public async Task PublishAsync(HealthReport report, CancellationToken cancellationToken)
```

Comment:
please decorate public methods

Reply:
Class is Internal and the interface used for accessing it provides documentation. Do you think we should add something more here?

Reply:
Didn't realise the class was internal.
It's probably okay as is.